xscen/aggregate.py

```diff
@@ -334,36 +335,70 @@ def spatial_mean(
     xarray.Dataset.mean, xarray.Dataset.interp, xesmf.SpatialAverager
     """
     kwargs = kwargs or {}
+    if method == "mean":
+        warnings.warn(
+            "xs.spatial_mean with method=='mean' is deprecated and will be abandoned in a future release. Use method=='cos-lat' instead for a more robust but similar method.",
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
+    elif method == "interp_coord":
+        warnings.warn(
+            "xs.spatial_mean with method=='interp_coord' is deprecated. Use method=='interp_centroid' instead.",
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
+        method = "interp_centroid"
 
     # If requested, call xscen.extraction.clisops_subset prior to averaging
     if call_clisops:
         ds = clisops_subset(ds, region)
 
+    if method == "cos-lat":
+        if "latitude" not in ds.cf.coordinates:
+            raise ValueError(
+                "Could not determine the latitude name using CF conventions. "
+                "Use kwargs = {lat: str} to specify the coordinate name."
+            )
+
+        if "units" not in ds.cf["latitude"].attrs:
+            logger.warning(
+                f"{ds.attrs.get('cat:id', '')}: Latitude does not appear to have units. Make sure that the computation is right."
+            )
+        elif ds.cf["latitude"].attrs["units"] != "degrees_north":
+            logger.warning(
+                f"{ds.attrs.get('cat:id', '')}: Latitude units is '{ds.cf['latitude'].attrs['units']}', expected 'degrees_north'. "
+                f"Make sure that the computation is right."
+            )
+
+        if (ds.cf["longitude"].min() < -160) & (ds.cf["longitude"].max() > 160):
```

Comment:
```suggestion
        if (
            ((ds.cf["longitude"].min() < -160) & (ds.cf["longitude"].max() > 160))
            or ((ds.cf["longitude"].min() < 20) & (ds.cf["longitude"].max() > 340))
        ):
```
To catch the other edge case.